lang/image.bzl

```diff
@@ -39,6 +39,13 @@ def _reference_dir(ctx):
   # /app/bar/baz/blah.runfiles/foo
   return "/".join([_runfiles_dir(ctx), ctx.workspace_name])
 
+# The special "external" directory which is an alternate way of accessing
+# other repositories.
+def _external_dir(ctx):
+  # For @foo//bar/baz:blah this would translate to
+  # /app/bar/baz/blah.runfiles/foo/external
```

Comment:
foo/external -> external/foo right?

by the way did you see https://github.com/bazelbuild/bazel/issues/1262

Reply:
Depends if `@foo` is ourself or not, which I'll admit wasn't super clear to me when I first lifted this from one of these other functions (`_reference_dir` IIRC).  From `@foo` a `@bar` reference would be: `...runfiles/foo/external/bar` == `...runfiles/bar`.